packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_multiplier.js

```diff
@@ -0,0 +1,227 @@
+
+        if (!market_block || !trade_type_block) {
+            return;
+        }
+
+        this.selected_symbol = market_block.getFieldValue('SYMBOL_LIST');
+        this.selected_trade_type_category = trade_type_block.getFieldValue('TRADETYPECAT_LIST');
+        this.selected_trade_type = trade_type_block.getFieldValue('TRADETYPE_LIST');
+        this.selected_multiplier = this.getFieldValue('MULTIPLIERTYPE_LIST');
+
+        const is_load_event = /^dbot-load/.test(event.group);
+
+        if (event.type === Blockly.Events.BLOCK_CREATE && event.ids.includes(this.id)) {
+            if (is_load_event) {
+                // Do NOT touch any values when a strategy is being loaded.
+                this.updateMultiplierInput(false);
+            } else {
+                this.updateMultiplierInput(true);
+            }
+        } else if (event.type === Blockly.Events.BLOCK_CHANGE) {
+            if (is_load_event) {
+                if (event.name === 'TRADETYPE_LIST') {
+                    this.updateMultiplierInput(false);
+                }
+            } else if (event.blockId === this.id) {
+                switch (event.name) {
+                    case 'MULTIPLIERTYPE_LIST': {
+                        this.updateMultiplierInput(false);
+                        break;
+                    }
+                    default:
+                        break;
+                }
+            } else if (event.name === 'SYMBOL_LIST' || event.name === 'TRADETYPE_LIST') {
+                this.updateMultiplierInput(true);
+            }
+        } else if (event.type === Blockly.Events.END_DRAG && event.blockId === this.id) {
+            // Ensure this block is populated after initial drag from flyout.
+            if (!this.selected_multiplier) {
+                const fake_creation_event = new Blockly.Events.Create(this);
+                fake_creation_event.recordUndo = false;
+                Blockly.Events.fire(fake_creation_event);
+            }
+        }
+    },
+    updateMultiplierInput(should_use_default_value) {
+        const { contracts_for } = ApiHelpers.instance;
+
+        if (this.selected_trade_type === 'multiplier') {
+            contracts_for.getMultiplierRange(this.selected_symbol, this.selected_trade_type).then(multiplier_range => {
+                if (multiplier_range.length > 0) {
+                    const multiplier_list_dropdown = this.getField('MULTIPLIERTYPE_LIST');
+                    const multiplier_options = multiplier_range.map(value => {
+                        const option = value.toString();
+                        return [option, option];
+                    });
+
+                    multiplier_list_dropdown.updateOptions(multiplier_options, {
+                        default_value: should_use_default_value ? undefined : multiplier_list_dropdown.getValue(),
+                    });
+                }
+            });
+        } else if (this.parentBlock_?.type === 'trade_definition') {
```

Comment:
```suggestion
        } else if (this.isDescendantOf('trade_definition')) {
```
i'm sorry, i didn't think of that before suggesting a change the first time. This check should be done recursively, because in many cases our users have different logical operators, so a `Duration` or `multipliers` blocks can be inside logical operators or functions so check would not work. We have a method for recursive check.